Adding tests to validate anomoly detection (#12)
avohq/avo_audit: README.md

# Avo Audit

> Avo audit helps monitor anomolies in your raw event schema, by setting up tests using `test_avo_audit_detect_event_count_anomoly` you will be able to constantly monitor for any unexpected spike or drop in volume for any event.

Reviewer: ```suggestion
Avo audit helps monitor anomalies in your raw event schema, by setting up tests using `test_avo_audit_detect_event_count_anomoly` you will be able to constantly monitor for any unexpected spike or drop in volume for any event.
```

Avo audit helps you detect issues in your raw analytics event tables by querying a sample of the raw events and comparing them against each other.

> This allows you to find discrepancies between events and properties that should look the same, but don’t.

Reviewer 2: Should we update the intro text here to better describe the current functionality?

# Macros

> ### test_avo_audit_detect_event_count_anomoly

Reviewer: ```suggestion
### test_avo_audit_detect_event_count_anomaly
```

      identifier="raw_event_table"
) -%}

> {{ avo_audit.test_avo_audit_detect_event_count_anomoly(

Reviewer: ```suggestion
{{ avo_audit.test_avo_audit_detect_event_count_anomaly(
```

    relation=raw_event_relation,
    event_name_column="event",
    event_date_column="ts",
    event_source_column="client",
    end_date="2021-11-25",
    n_days=15
) }}
```

> This will return a view where each row represents a event with a spike or drop in data on any day in the past 15 days.

Reviewer: ```suggestion
This will return a view where each row represents an event with a spike or drop in data on any day in the past 15 days.
```

> This can also be set up as an automatic test to be run each day against the raw data to make sure you catch any anomoly as soon as possible.

Reviewer: ```suggestion
This can also be set up as an automatic test to be run each day against the raw data to make sure you catch any anomaly as soon as possible.
```

```
// Example.yml

models: 
  - name: NAME_OF_RAW_TABLE
    tests:

>       - avo_audit.avo_audit_detect_event_count_anomoly:

Reviewer: ```suggestion
      - avo_audit.avo_audit_detect_event_count_anomaly:
```

          event_name_column: <NAME_OF_EVENT_NAME_COLUMN>
          event_date_column: <NAME_OF_DATE_COLUMN>
          event_source_column: <NAME_OF_SOURCE_COLUMN> // ios / android / web 
```

> This test will by default check the past 15 days from yesterday, and it will fail if any rows are returned which indicates an anomoly in event volume data for any event in your raw data.

Reviewer: ```suggestion
This test will by default check the past 15 days from yesterday, and it will fail if any rows are returned which indicates an anomaly in event volume data for any event in your raw data.
```

    )
}}
```
> This macro is a work in progress and is intended to help joining together tables of single events into one table to automatically include all events.

> test_avo_audit_detect_event_count_anomoly can also instead be applied as a test on all event tables that you want to monitor instead.

Reviewer: ```suggestion
test_avo_audit_detect_event_count_anomaly can also instead be applied as a test to all event tables that you want to monitor instead.
```

    )
}}
```
> This macro is a work in progress and is intended to help joining together tables of single events into one table to automatically include all events.
test_avo_audit_detect_event_count_anomoly can also instead be applied as a test on all event tables that you want to monitor instead.
# Coming soon
## Issue types

### Issues available in the alpha version:  

**Property sometimes missing** – detects absence of event properties across tracking calls.  
>_Property onboarding status is not sent with 30% of the App Opened event._ 

**Property type mismatch** – detects inconsistent event property types across tracking calls.  

> >_Inconsistent type of property onboarding status on event App Opened: int (84%), string (16%)._

Reviewer 2: Is this true?
